lib/spack/spack/version.py

```diff
@@ -643,6 +643,19 @@ def highest(self):
         else:
             return self[-1].highest()
 
+    def highest_numeric(self, fallback_all=False):
```

Comment:
Should we return the highest numeric or the preferred version?

Reply:
Good idea, actually `highest_numeric(True)` is *hopefully* the preferred one in default settings...